compiler/src/main/java/org/robovm/compiler/AppCompiler.java

```diff
@@ -241,10 +244,33 @@ private boolean compile(Executor executor, ClassCompilerListener listener,
                     compileQueue.add(depClazz);
                 }
             }
+            addMetaInfImplementations(config.getClazzes(), clazz, compiled, compileQueue);
         }
         return result;
     }
 
+    static void addMetaInfImplementations(Clazzes clazzes, Clazz clazz, Set<Clazz> compiled, Set<Clazz> compileQueue) throws IOException {
+        String metaInfName = "META-INF/services/" + clazz.getClassName();
+        for (InputStream is : clazzes.loadResources(metaInfName)) {
```

Comment:
If parsing one of the streams goes wrong, the remaining streams aren't closed. May get cleaned up in finalizes eventually, but given enough heap that might never happen and we could run out of file handles.